homeassistant/components/calendar/__init__.py

```diff
@@ -135,6 +163,8 @@ def cleanup(self):
 
     def update(self):
         """Search for the next event."""
+        self._event_list = self.data.event_list
```

Comment:
We should not cache event list in memory. We should generate them on the fly and then return from API.

Reply:
What do you mean by "on the fly" ? Run `update` (or `async_update`) for each device and get the output ?
You said that because the data are already cached with the `Throttle` decorator, don't you ?